code/game/objects/items/weapons/cards_ids.dm

```diff
@@ -329,3 +329,27 @@ var/const/NO_EMAG_ACT = -50
 
 /obj/item/card/id/blankwhite
 	icon_state = "id_blankwhite"
+
+/obj/item/card/id/randomassistant
+	var/csex
+
+/obj/item/card/id/randomassistant/examine(mob/user)
+	if("\[UNSET\]" != blood_type) // we have to call the parent only if the variables are loaded or else these variables will only appear after the first examine
+		return ..()
+	dna_hash = sha1("A"+registered_name) // something is subtly wrong with these IDs
+	fingerprint_hash = md5("A"+registered_name)
+	blood_type = pick(GLOB.blood_types)
```

Comment:
Do these really need to be in examine code?

Reply:
I mostly did it because I wanted to try to create an oldified version of the directional sprites, but felt it would be too expensive to place them on every ID. before I realized sprite code is more strenuous than I thought, I coded these things into examine as they are not visible from anything but examine, and so would not need to be generated for ID cards that are never examined.